docs/icicle/introduction.md

```diff
@@ -1,8 +1,16 @@
 # Getting started with ICICLE
 
+This guide is oriented towards developers who want to start writing code with the ICICLE libraries. If you just want to run your existing ZK circuits on GPU refer to [this guide](./integrations.md#using-icicle-integrations) please.
+
 ## ICICLE repository overview
 
-The ICICLE repository has three main sections, each section is independent from the other.
+![ICICLE API overview](../../static/img/apilevels.png)
+
+The diagram above displays the general architecture of ICICLE and the API layers that exist. The CUDA API, which we also call ICICLE Core, is the lowest level and is comprised of CUDA kernels which implement all primitives such as MSM as well as C++ wrappers which expose these methods for different curves.
+
+ICICLE Core compiles into a static library. This library can be used with our official Golang and Rust wrappers or you can implement a wrapper for it in any language.
+
+Based on this dependency architecture, the ICICLE repository has three main sections, each of which is independent from the other.
 
 - ICICLE core
 - ICICLE Rust bindings
```

Comment:
Github won't let me comment on line 23...

"ICICLE Core would typically be compiled into a ~~shared~~static library and used in a third party language such as Rust, Golang or Python."